login page and product page added

Psahu1296 opened the pull request:
1. login page and product page added.
2. unit test cases added.

Psahu1296 committed:
login page and product page added

Psahu1296 committed:
updated login methods

hhkk28 commented on src/App.tsx:
```diff
@@ -1,13 +1,23 @@
+import Products from "./components/Products/Products";
 
 export default function App() {
+  const [isLoggedin, setIsLoggedin] = useState<boolean>(false);
```
What happens if the user has no valid credentials? Please include the error handling.

hhkk28 commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,104 @@
+}
+
+const Login = ({ loginFlagHandler }: LoginProps) => {
+  const [user, setUser] = useState<string>("");
```
Please refactor the form handling with useReduce instead of using multiple useState.

Psahu1296 replied:
addressed the comment.

hhkk28 commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,104 @@
+          Sign In
+        </h1>
+        <div className="input_wrapper">
+          <label data-testid="user-label" htmlFor="username">
```
`htmlFor` attribute does not work without `id` attribute in the corresponding input element. Please update the code.

Psahu1296 replied:
addressed the comment.

hhkk28 commented on src/components/Login/Login.spec.tsx:
```diff
@@ -0,0 +1,32 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
```
There is no test to check the error. please write tests for the failing and negative conditions.

Psahu1296 replied:
addressed the comment.

Psahu1296 committed:
fix: infinite product api call.
 - Added dependency array to products api useEffect.
 - added negative/error unit test cases.
 - used useReducer over multiple useState.

Prathamesh9997 commented on src/components/Products/Products.tsx:
```diff
@@ -0,0 +1,46 @@
+
+import React, { useEffect, useState } from "react";
+
+interface ProductParams {
```
Create separate file (types.ts) to store all the interfaces and types of the specific component.

Psahu1296 replied:
addressed the comment.

Prathamesh9997 commented on src/components/Products/Products.tsx:
```diff
@@ -0,0 +1,46 @@
+
+export default Products;
+
+const Product = (params: ProductParams) => {
```
Create a separate component for Product. Its good practice to have one component per file.

Psahu1296 replied:
addressed the comment.

Prathamesh9997 commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,144 @@
+const URL = "https://fakestoreapi.com/auth/login";
+const userData = ["User123", "abcd1234"];
+
+interface LoginProps {
```
Same here. User separate file for interfaces, enums, types etc.

Psahu1296 replied:
addressed the comment.

Prathamesh9997 commented on src/components/Login/Login.css:
```diff
@@ -0,0 +1,84 @@
+    font-size: 24px;
+    font-weight: 700;
+    margin-bottom: 15px;
+    /* letter-spacing: 18px; */
```
Never commit commented out code. Always remove dead codes.

Psahu1296 replied:
addressed the comment.

Prathamesh9997 commented on src/App.tsx:
```diff
@@ -1,13 +1,23 @@
+import { useEffect, useState } from "react";
 
 import styles from "./App.module.css";
+import Login from "./components/Login/Login";
```
Use absolute path instead of relative. eg. Instead of writing "./components/" always use "@/components/".

In tsconfig.json paths are already mentioned. You can add absolute path for any folder. Lets say you want to add absolute path for public folder then simply add path "@public/*": ["./public/*"] in tsconfig.json file

Psahu1296 replied:
addressed the comment.

Prathamesh9997 commented on src/App.tsx:
```diff
@@ -1,13 +1,23 @@
 
 import styles from "./App.module.css";
+import Login from "./components/Login/Login";
+import Products from "./components/Products/Products";
```
Here you are importing Login and Products from components folder, so import it like this: 
import {Login, Products} from "@/components";

We will connect on this.

Psahu1296 replied:
addressed the comment.

Psahu1296 committed:
refactor: refactored code to export single component from one file.

 - new separated product component added with unit test.
 - removed redundant code.
 - added types file for interface, types and enums.
